src/gui/guiFormSpecMenu.h

```diff
@@ -331,6 +331,13 @@ class GUIFormSpecMenu : public GUIModalMenu
 	u16 m_selected_amount = 0;
 	bool m_selected_dragging = false;
 	ItemStack m_selected_swap;
+	int m_held_button = -1;
+	bool m_shift_craft = false;
+
+	u16 m_left_drag_amount;
+	ItemStack m_left_drag_stack;
+	std::map<GUIInventoryList::ItemSpec *, ItemStack> m_left_drag_stacks;
```

Comment:
`GUIInventoryList::ItemSpec` must be freed within `~GUIFormSpecMenu`.
Possible solution 1: use `std::unique_ptr` to ensure that they're deleted with the map
Suggested solution 2: use `std::vector<std::pair<GUIInventoryList::ItemSpec, ItemStack>>` or a custom struct to hold both values.